### ISSUE TITLE
[BUG] Masking Combinator -- NotFoundLookupError

### ISSUE THREAD

**Issue by mirkoklukas** [genjax-community/genjax#761]
**Describe the bug**
I would have expected the code below to run:

```
key = keysplit(key)
masked = genjax.masking_combinator(genjax.bernoulli)
genjax.Map(masked, in_axes=(0,(0,))).simulate(key, ( 
    jnp.array([1,1,0]), 
    (logit(jnp.array([0.5,0.5,0.5])),),
))
```

**To Reproduce**
Run code above.

**Expected behavior**
Return a masked trace. I would have expeted a similar behaviour than this:

```
key, keys = keysplit(key, 1, 3)
masked = genjax.masking_combinator(genjax.bernoulli)
tr = vmap(masked.simulate, in_axes=(0, (0,(0,))) )(keys, ( 
    jnp.array([1,1,0]), 
    (logit(jnp.array([0.5,0.5,0.5])),),
))
```

**Screenshots**

**Execution environment (please complete the following information):**

* GCP, Ubuntu (?)

**Additional context**
stack trace:

```

{
	"name": "NotFoundLookupError",
	"message": "For function `map_combinator`, `(MaskingCombinator(inner=TFPDistribution(make_distribution=<function <lambda> at 0x7f7422466ca0>)),)` could not be resolved.",
	"stack": "---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/function.py:421, in Function._resolve_method_with_cache(self, args, types)
    420 try:
--> 421     return self._cache[types]
    422 except KeyError:

KeyError: (<class 'genjax._src.generative_functions.combinators.masking_combinator.MaskingCombinator'>,)

During handling of the above exception, another exception occurred:

NotFoundLookupError                       Traceback (most recent call last)
File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/function.py:341, in Function.resolve_method(self, target)
    339 try:
    340     # Attempt to find the method using the resolver.
--> 341     signature = self._resolver.resolve(target)
    342     method = signature.implementation

File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/resolver.py:168, in Resolver.resolve(self, target)
    166 if len(candidates) == 0:
    167     # There is no matching signature.
--> 168     raise NotFoundLookupError(f\"`{target}` could not be resolved.\")
    169 elif len(candidates) == 1:
    170     # There is exactly one matching signature. Success!

NotFoundLookupError: `(MaskingCombinator(inner=TFPDistribution(make_distribution=<function <lambda> at 0x7f7422466ca0>)),)` could not be resolved.

During handling of the above exception, another exception occurred:

NotFoundLookupError                       Traceback (most recent call last)
Cell In[23], line 3
      1 key = keysplit(key)
      2 masked = genjax.masking_combinator(genjax.bernoulli)
----> 3 genjax.Map(masked, in_axes=(0,(0,))).simulate(key, ( 
      4     jnp.array([1,1,0]), 
      5     (logit(jnp.array([0.5,0.5,0.5])),),
      6 ))

File ~/workspace/genjax/src/genjax/_src/core/datatypes/generative.py:1518, in LanguageConstructor.__call__(self, *args, **kwargs)
   1517 def __call__(self, *args, **kwargs):
-> 1518     return self.constructor(*args, **kwargs)

File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/function.py:398, in Function.__call__(self, *args, **kw_args)
    397 def __call__(self, *args, **kw_args):
--> 398     method, return_type = self._resolve_method_with_cache(args=args)
    399     return _convert(method(*args, **kw_args), return_type)

File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/function.py:427, in Function._resolve_method_with_cache(self, args, types)
    424     args = Signature(*(resolve_type_hint(t) for t in types))
    426 # Cache miss. Run the resolver based on the arguments.
--> 427 method, return_type = self.resolve_method(args)
    428 # If the resolver is faithful, then we can perform caching using the types
    429 # of the arguments. If the resolver is not faithful, then we cannot.
    430 if self._resolver.is_faithful:

File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/function.py:350, in Function.resolve_method(self, target)
    348 except NotFoundLookupError as e:
    349     e = self._enhance_exception(e)  # Specify this function.
--> 350     method, return_type = self._handle_not_found_lookup_error(e)
    352 return method, return_type

File /opt/conda/envs/dcolmap/lib/python3.11/site-packages/plum/function.py:359, in Function._handle_not_found_lookup_error(self, ex)
    354 def _handle_not_found_lookup_error(
    355     self, ex: NotFoundLookupError
    356 ) -> Tuple[Callable, TypeHint]:
    357     if not self.owner:
    358         # Not in a class. Nothing we can do.
--> 359         raise ex
    361     # In a class. Walk through the classes in the class's MRO, except for this
    362     # class, and try to get the method.
    363     method = None

NotFoundLookupError: For function `map_combinator`, `(MaskingCombinator(inner=TFPDistribution(make_distribution=<function <lambda> at 0x7f7422466ca0>)),)` could not be resolved."
}
```

**Comment by femtomc** [comment]
I think the issue here is that `MapCombinator` expects a `JAXGenerativeFunction` and `MaskingCombinator` is not explicitly marked as a subtype of `JAXGenerativeFunction`.

But this is an okay fix to make for now.

**Comment by sritchie** [comment]
Closing since this is fixed in 0.4.0, says @femtomc .
